tasks/main.yml

```diff
@@ -6,43 +6,21 @@
     virtualenv: "{{ os_container_clusters_venv or omit }}"
   become: "{{ os_container_clusters_venv == None }}"
 
-- name: Check whether container cluster templates exist
-  shell: >
-    {{ os_container_clusters_venv ~ '/bin/' if os_container_clusters_venv else '' }}openstack
-    {% for auth_name, auth_value in os_container_clusters_auth.items() %}
-    --os-{{ auth_name | replace('_', '-') }}='{{ auth_value }}'
-    {% endfor %}
-    {% if os_container_clusters_cacert is defined %}
-    --os-cacert {{ os_container_clusters_cacert }}
-    {% endif %}
-    --os-interface=public
-    coe cluster template show {{ item.name }} >/dev/null
+- name: Ensure container cluster templates exist
+  openstack.cloud.coe_cluster_template:
+    auth: "{{ os_container_clusters_auth }}"
+    auth_type: "{{ os_container_clusters_auth_type }}"
+    labels: "{{ item.labels | default(omit) }}"
+    external_network_id: "{{ item.external_network_id | default('external') }}"
+    master_flavor_id: "{{ item.master_flavor | default('m1.small') }}"
+    flavor_id: "{{ item.flavor | default('m1.small') }}"
```

Comment:
I'd prefer to not provide defaults for these. If they are mandatory then let's fail.